src/inmanta/server/server.py

```diff
@@ -128,15 +129,21 @@ async def get_server_status(self) -> StatusResponse:
                 "Is inmanta installed? Use setuptools install or setuptools dev to install."
             )
 
-        slices = []
-        for slice_name, slice in self._server.get_slices().items():
+        async def collect_for_slice(slice_name: str, slice: protocol.ServerSlice) -> SliceStatus:
             try:
-                slices.append(SliceStatus(name=slice_name, status=await slice.get_status()))
+                return SliceStatus(name=slice_name, status=await asyncio.wait_for(slice.get_status(), 0.1))
+            except TimeoutError:
+                return SliceStatus(name=slice_name, status={"error": "timeout after 100ms"})
```

Comment:
Should we have the frontend retry regularly if this occurs? Also for handling these new error responses in general, should we create a frontend ticket?

Reply:
```suggestion
                return SliceStatus(name=slice_name, status={"error": f"timeout on data collection for {slice_name}, consult the server log for additional information"})
```

Reply:
I discussed it and this is what we came up with

Reply:
So no retries? I assume this is an exceptional case and indicates something is wrong?

Reply:
Indeed. (and they naturally retry every 5 seconds)